### PR TITLE
Update token feature

### PR DESCRIPTION
**Pull request by oddevan**
Continuation of PR #18 adding the phpDocBlocks and tests asked for in that PR.

This PR has been filed in the hopes of getting this feature into a shipping version of the library sooner, as I have multiple projects that could use it.

From @solverat's original:

> This PR:
>
> - allows you to exchange a short-lived access token with a long-lived access-token (which should fix Not working #15)
> - allows you to refresh a long-lived token
> - fixes two annotations return types
> - fixes wrong Exception class (IdentityProviderException => InstagramIdentityProviderException)

[API Reference](https://developers.facebook.com/docs/instagram-basic-display-api/guides/long-lived-access-tokens)

### CODE REVIEW

**Comment by solverat** [comment]
@oddevan Thanks for the finishing!

**Comment by KevinBerkhof** [comment]
Nice job (both of you!). Hopefully @stevenmaguire can add this into a release soon. ?

File: src/Grant/IgExchangeToken.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace League\OAuth2\Client\Grant;
+
+class IgExchangeToken extends AbstractGrant
```
Comment (shadowhand) [inline src/Grant/IgExchangeToken.php]: Let's rename this `InstagramExchangeToken` for clarity.

File: src/Grant/IgRefreshToken.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace League\OAuth2\Client\Grant;
+
+class IgRefreshToken extends AbstractGrant
```
Comment (shadowhand) [inline src/Grant/IgRefreshToken.php]: Let's rename this `InstagramRefreshToken` for clarity.

**Review by shadowhand — CHANGES_REQUESTED** [review]
Just two minor class renames, and then I will merge and release this. Thanks!

**Comment by oddevan** [comment]
@shadowhand Got it!

**Review by shadowhand — APPROVED** [review]

**Comment by shadowhand** [comment]
Released version 3.1.0 moments ago.

File: src/Grant/IgExchangeToken.php (hunk 1 above)
Comment (shadowhand) [reply src/Grant/IgExchangeToken.php]: Turns out this naming was consistent with the expectations of `GrantFactory` 😓 Sorry about that! I renamed both in 06aa2721c33bd8fc512798af301fb405dda79542

**Comment by solverat** [comment]
Nice! 👍

**Pull request merged**